src/search3_rpc.erl

```diff
@@ -343,3 +344,28 @@ post(Action, Request, RequestType, ResponseType) ->
         {error, Reason} ->
             {error, Reason}
     end.
+
+get_endpoints() ->
+    Host = config:get("search3", "service_url", "localhost"),
+    case Host of
+        ":discover" ->
+            discover_endpoints();
+        _ ->
+            Port = config:get_integer("search3", "service_port", 8443),
+            to_conn_str(Host, Port)
+    end.
+
+discover_endpoints() ->
+    DNSName = config:get("search3", "service_dns"),
+    case inet_res:getbyname(DNSName, srv) of
+        {ok, #hostent{h_addr_list = AddrList}} ->
+            lists:map(fun({_Priority, _Weight, Port, Host}) ->
+                to_conn_str(Host, Port)
+            end, AddrList);
+        {error, Reason} ->
+            erlang:error({service_lookup_failed, Reason})
+    end.
+
+to_conn_str(Host, Port) ->
+    Str = io_lib:format("http://~s:~b/Search/", [Host, Port]),
```

Comment:
once tls is enabled (meaning we know how to get the certificate files into a kube environment we can change this https